arch/x86/kvm/pkvm/pgtable.h

```diff
@@ -191,4 +191,28 @@ int pkvm_pgtable_walk(struct pkvm_pgtable *pgt, unsigned long vaddr,
 int pkvm_pgtable_map(struct pkvm_pgtable *pgt, unsigned long vaddr,
 		     unsigned long phys, unsigned long size, u64 prot);
 
+/*
+ * pkvm_pgtable_unmap() - Remove a vaddr to phys mapping from a pkvm pgtable.
+ *
+ * @pgt:	The page table to remove mapping.
+ * @vaddr:	The virtual address of the mapping to be removed.
+ * @phys:	The desired physical address for the unmap walker to verify.
+ * @size:	The memory size to unmap.
+ *
+ * The @phys is used by the unmap walker to verify if the physical address in the
+ * removed mapping is desired or not. If phys is INVALID_PAGE, this verification
+ * will not be performed.
```

Comment:
Doesn't mention about aligning to PAGE_SIZE?

Reply:
I will mention that.

For the map/unmap/lookup API, as they are all using pkvm_pgtable_walk(), the alignment for vaddr/size are following what the pkvm_pgtable_walk() does. So maybe we can say:
"The alignment for the @vaddr and @size follows the pkvm_page_walk(), and the @phys is aligned down to the PAGE_SIZE"

Reply:
> "The alignment for the @vaddr and @SiZe follows the pkvm_page_walk()

Actually IMHO it is not a problem to duplicate this description for each function (so that the reader doesn't need to go read the pkvm_page_walk() description). This description is not very long, and this idea of page alignment is rather straightforward and intuitive.